portfolio/src/main/java/com/google/sps/data/BookServiceClient.java

```diff
@@ -0,0 +1,66 @@
+package com.google.sps.data;
+ 
+import com.google.sps.data.Book;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
+import java.lang.NullPointerException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLEncoder;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+ 
+// Given a book name, fetches book info json by calling Google books api and returns as string.
+public class BookServiceClient {
+ 
+  public static String getBookInfo(String bookName) throws Exception {
+    if(validate(bookName)){
+        String encodedBookName = null;
+        encodedBookName = URLEncoder.encode(bookName, "UTF-8");
+        String allResults = queryBooksAPI(encodedBookName);
+        return getTopResult(allResults);
+    }
+    return null;
+  }
+ 
+  private static String getTopResult(String booksApiResults){
+    //test to check that it took the right balue off
+    JSONObject jsonObject = new JSONObject(booksApiResults);
+    JSONArray allItems = jsonObject.getJSONArray("items");
+    JSONObject firstItem = allItems.getJSONObject(0);//returns top result'
+    return firstItem.toString();
+  }
+
+  private static String queryBooksAPI(String bookName) throws Exception {
```

Comment:
Consider writing short descriptions of what the methods do, since some of them are not 100% self-explanatory! 
https://google.github.io/styleguide/javaguide.html#s7-javadoc